pkg/theiactl/commands/utils.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func CreateK8sClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var err error
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return clientset, nil
+}
+
+func PolicyRecoPreCheck(clientset *kubernetes.Clientset) error {
+	// Check the deployment of Spark Operator in flow-visibility ns
+	pods, err := clientset.CoreV1().Pods(flowVisibilityNS).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "app.kubernetes.io/instance=policy-reco,app.kubernetes.io/name=spark-operator",
+	})
+	if err != nil {
+		return fmt.Errorf("error %v when finding the policy-reco-spark-operator Pod, please check the deployment of the Spark Operator", err)
+	}
+	if len(pods.Items) < 1 {
+		return fmt.Errorf("can't find the policy-reco-spark-operator Pod, please check the deployment of the Spark Operator")
+	}
+	CheckClickHousePod(clientset)
+	return nil
+}
+
+func CheckClickHousePod(clientset *kubernetes.Clientset) error {
+	// Check the ClickHouse deployment in flow-visibility namespace
+	pods, err := clientset.CoreV1().Pods(flowVisibilityNS).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "app=clickhouse",
+	})
+	if err != nil {
+		return fmt.Errorf("error %v when finding the ClickHouse Pod, please check the deployment of the ClickHouse", err)
+	}
+	if len(pods.Items) < 1 {
+		return fmt.Errorf("can't find the ClickHouse Pod, please check the deployment of ClickHouse")
+	}
+	return nil
+}
+
+func getServicePort(clientset *kubernetes.Clientset, name string) (int32, error) {
+	var servicePort int32
+	service, err := clientset.CoreV1().Services(flowVisibilityNS).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return servicePort, fmt.Errorf("error %v when finding Service %s, please check the deployment of the ClickHouse", err, name)
```

Comment:
If this function would be used for other services, we shouldn't mention `Clickhouse` here? Otherwise name should reflect that this is Clickhouse-specific.

Reply:
As currently no other service requires this function, rename it to Clickhouse-specific.